src/daft-plan/src/logical_ops/join.rs

```diff
@@ -167,6 +169,37 @@ impl Join {
         }
     }
 
+    fn rename_join_keys(
+        left_exprs: Vec<Arc<Expr>>,
+        right_exprs: Vec<Arc<Expr>>,
+    ) -> (Vec<Arc<Expr>>, Vec<Arc<Expr>>) {
+        left_exprs
+            .into_iter()
+            .enumerate()
+            .map(|(i, left_expr)| {
+                let right_expr = &right_exprs[i];
```

Comment:
Small nit, but instead of enumerating and indexing here, you can do something along the lines of `left_exprs.into_iter().zip(right_exprs)`

Reply:
Good point, just pushed a change on these lines.